app/models/system-state.server.ts

```diff
@@ -0,0 +1,23 @@
+import { StateEnumType } from '@prisma/client';
+
+import { prisma } from '~/db.server';
+
+import type { SystemState } from '@prisma/client';
+
+export function getIsReconciliationNeeded(): Promise<SystemState['reconciliationNeeded']> {
+  return prisma.systemState
+    .findUnique({
+      select: { reconciliationNeeded: true },
+      where: { unique: StateEnumType.unique },
+    })
+    .then((data) => data!.reconciliationNeeded);
```

Comment:
The use of `data!.` here makes me a bit nervous here, since it will bring down the system pretty quick if it fails at runtime.  I get that it shouldn't fail, but...

Given that this *could* always return `undefined`, could we wrap this access in a `try/catch` and return `true` if `data` is missing at runtime?  Doing so would force the reconciler to call `setIsReconciliationNeeded` soon and fix the situation so that the record does get added.

Reply:
Thinking about this, you might just solved my e2e testing issue